python/sglang/srt/mem_cache/sparsity/algorithms/base_algorithm.py

```diff
@@ -0,0 +1,475 @@
+import logging
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import TYPE_CHECKING, Any, Dict, Optional
+
+import torch
+
+if TYPE_CHECKING:
+    from sglang.srt.model_executor.forward_batch_info import ForwardBatch
+
+logger = logging.getLogger(__name__)
+
+
+class SparseMode(Enum):
+    """Sparse attention granularity mode."""
+
+    PAGE_WISE = "page_wise"
+    TOKEN_WISE = "token_wise"
+    ORIGINAL_WISE = "original_wise"
+
+
+class BaseSparseAlgorithm(ABC):
```

Comment:
let's aim for to get the algorithm part under `/mem_cache/sparsity/algorithm` as a separated PR to get merged first, with a more detailed review, proper explanation and testing